src/pynwb/__init__.py

```diff
@@ -354,6 +354,24 @@ def get_sum(self, a, b):
     return __TYPE_MAP.get_dt_container_cls(neurodata_type, namespace)
 
 
+@docval({'name': 'path', 'type': str, 'doc': 'the neurodata_type to get the NWBContainer class for'},
+        is_method=False)
+def _get_backend(path: str):
+    from hdmf_zarr import NWBZarrIO
+
+    backend_io_classes = [NWBHDF5IO, NWBZarrIO]
+    backend_options = [b for b in backend_io_classes if b.can_read(path=path)]
+    if len(backend_options) == 0:
+        warn(f"Could not find an IO to read the file '{path}'." 
+             f"This may be due to an older file version or invalid file." 
+             f"Defaulting to NWBHDF5IO.", UserWarning)
+        return NWBHDF5IO
+    elif len(backend_options) > 1:
+        raise ValueError(f"Multiple backends found for file '{path}': {backend_options}")
+    else:
+        return backend_options[0]
```

Comment:
I have this temporary private method to start testing support for validation of Zarr NWB files as well. This can eventually be replaced by methods from the `read_nwb` PR